app/services/workflow_link_service.rb

```diff
@@ -11,7 +11,7 @@ def initialize(workflow_id)
   end
 
   def link(tag_attrs)
-    validate_approver_groups(Workflow.find(workflow_id).group_refs)
+    validate_and_update_approver_groups(Workflow.find(workflow_id))
```

Comment:
Why do we need to update `workflow` here? We only validate before, otherwise raise an exception

Reply:
It gives us a chance to refresh the group name which is now also stored in our DB.